pkg/cmd/start/start.go

```diff
@@ -113,6 +119,41 @@ func runGather(operator *controller.GatherJob, cfg *controllercmd.ControllerComm
 		protoConfig.ContentType = "application/vnd.kubernetes.protobuf"
 
 		ctx, cancel := context.WithTimeout(context.Background(), operator.Interval)
+
+		// before calling the Gather logic, this command must check the featuregate status.
+		configClient, err := configv1client.NewForConfig(clientConfig)
+		if err != nil {
+			klog.Exit("error: building client: %v", err)
+		}
+		configInformers := configv1informers.NewSharedInformerFactory(configClient, 10*time.Minute)
+
+		desiredVersion := os.Getenv("RELEASE_VERSION")
```

Comment:
What launches the gather job?  Having this match is important to get consistent feature gate behavior.

Reply:
This can be launched "manually" by the user (kubeadmin). Basically as an one-off operation - see https://docs.openshift.com/container-platform/4.12/support/remote_health_monitoring/remote-health-reporting-from-restricted-network.html. I think this is very edge case and I plan to remove it once the on-demand gathering is GA.